File: smoke_tests/go/sql_injection.go

package main

import (
	"database/sql"
	"fmt"
	"net/http"
)

func vulnerable(w http.ResponseWriter, r *http.Request) {
	userId := r.URL.Query().Get("id")
	db, _ := sql.Open("mysql", "user:password@/dbname")
	// Vulnerable to SQL Injection
	query := fmt.Sprintf("SELECT * FROM users WHERE id = '%s'", userId)
	db.Query(query)
}

File: smoke_tests/javascript/sql_injection.js

const express = require('express');
const mysql = require('mysql');
const app = express();

const connection = mysql.createConnection({
  host: 'localhost',
  user: 'root',
  password: 'password',
  database: 'db'
});

app.get('/user', (req, res) => {
  const userId = req.query.id;
  // Vulnerable to SQL Injection
  const query = "SELECT * FROM users WHERE id = '" + userId + "'";
  connection.query(query, (err, results) => {
    res.send(results);
  });
});

app.listen(3000);
